specs/Standards/Sould Bound Token/README.md

```diff
@@ -0,0 +1,235 @@
+    pub description: Option<String>, // free-form description
+    pub media: Option<String>, // URL to associated media, preferably to decentralized, content-addressed storage
+    pub media_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of content referenced by the `media` field. Required if `media` is included.
+    pub copies: Option<u64>, // number of copies of this set of metadata in existence when token was minted.
+    pub issued_at: Option<u64>, // When token was issued or minted, Unix epoch in milliseconds
+    pub expires_at: Option<u64>, // When token expires, Unix epoch in milliseconds
+    pub starts_at: Option<u64>, // When token starts being valid, Unix epoch in milliseconds
+    pub updated_at: Option<u64>, // When token was last updated, Unix epoch in milliseconds
+    pub extra: Option<String>, // anything extra the SBT wants to store on-chain. Can be stringified JSON.
+    pub reference: Option<String>, // URL to an off-chain JSON file with more info.
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+
+trait SBT {
+    /**********
+    * QUERIES
+    **********/
+
+    /// get the information about specific token ID
+    fn sbt(&self, token_id: TokenId) -> Option<Token>;
+
+    /// returns total amount of tokens minted by this contract
+    fn sbt_total_supply(&self) -> U64;
+
+    /// returns total supply of SBTs for a given owner
+    fn sbt_supply_for_owner(&self, account: AccountId);
+
+    /// Query for sbt tokens
+    fn sbt_tokens(&self, from_index: Option<U64>, limit: Option<u32>) -> Vec<Token>;
+
+    /// Query sbt tokens by owner
+    fn sbt_tokens_for_owner(
+        &self,
+        account: AccountId,
+        from_index: Option<U64>,
+        limit: Option<u32>,
+    ) -> Vec<Token>;
+
+    /// Optional. If the SBT implementaiton assures that one account can have maximum one SBT
+    /// the the following function should be implemented.
+    /// Returns Some(Token) if an `account` owns an SBT, otherwise returns None.
+    fn sbt_token_for_owner(&self, account: AccountId) -> Option<Token> {}
+
+    /**********
+    * TRANSACTIONS
+    **********/
+
+    /// creates a new, unique token and assigns it to the `receiver`.
+    #[payable]
+    fn sbt_mint(&mut self, metadata: TokenMetadata, receiver: AccountId);
+
+    /// sbt_recover reassigns all tokens from the old owner to a new owner,
+    /// and registers `old_owner` to a burned addresses registry.
+    /// Must be called by operator.
+    /// Must provide 5 miliNEAR to cover registry storage cost. Operator should
+    ///   put that cost to the requester (old_owner), eg by asking operation fee.
+    #[payable]
+    fn sbt_recover(&mut self, from: AccountId, to: AccountId);
+
+    /// sbt_renew will update the expire time of provided tokens.
+    /// `expires_at` is a unix timestamp (in seconds).
```

Comment:
Are sure that seconds is a good idea here? The NFT metadata standard mandates milliseconds and you can this here in lines 51-54. I propose to keep it consistent. I also suggest we use either milliseconds or nanoseconds, as these are provided as [functions](https://docs.rs/near-sdk/latest/near_sdk/env/index.html#functions) from near-sdk-rs. With NFTs, most of the old contracts just use nanoseconds instead of the standardized milliseconds, same will probably happen here. Esp. in case of nanoseconds `U64` or `U128` data type might be more appropriate.

Reply:
I prefer `seconds`, as the most natural, and good enough metric (unix timestamp measure, SI unit).
Personally, I think use of nano-seconds in NEAR is over-engineering. 
However, if the ecosystem settled on milliseconds, then we can use milliseconds.

Reply:
NEAR VM provides [`block_timestamp`](https://docs.rs/near-sdk/latest/near_sdk/env/fn.block_timestamp.html) in nanoseconds, so it is usually a no-brainer to use that value to compare it.

Reply:
`nanoseconds` are not user friendly. I don't believe anyone things in terms of nanoseconds when defining `expires_at`. Most of the people will think in terms of days or hours for things related to human. Seconds are a good Unix standard. Milliseconds are default in JS. 
Even if there are 100 blocks per second - the condition will be well defined. it's event the last transaction from the given block if using <= or the last transaction from the previous block if using < .
I don't believe anyone will like to go to nanosecond precision for things like `SBT.expires_at`. I other words, why anyone would put expires at: `April 7 2023 11:23:56.751209857`?
Also, worth to note that the block unit is a sequence number (1,2,3 ...). 

With that, I think `second` (as more human friendly) or `millisecond` (as JS default) are definitely more appropriate. My preference is `seconds`. However, if there is broader adoption for `milliseconds` then let's use milliseconds.

Reply:
I don't have a strong opinion on seconds/milliseconds/nanosecond for the expiration timestamp. There are pros and cons to each of them. Just make sure you align the whole document on using one selected option as currently `TokenMetadata` states that it is milliseconds precision.